kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/DependencyResolver.java

```diff
@@ -111,25 +113,28 @@ public ClusterResolutionResult resolveClusterRefs(VirtualKafkaCluster cluster, C
         Map<LocalRef<KafkaProtocolFilter>, KafkaProtocolFilter> filters = context.getSecondaryResources(KafkaProtocolFilter.class).stream()
                 .collect(ResourcesUtil.toByLocalRefMap());
         var clusterResolutionResults = virtualKafkaClusters.stream()
-                .map(cluster -> discoverProblemsAndBuildResolutionResult(cluster, ingresses, clusterRefs, filters, proxies)).collect(Collectors.toSet());
+                .map(cluster -> discoverProblemsAndBuildResolutionResult(cluster, ingresses, clusterRefs, filters, proxies, considerClusterConditions))
+                .collect(Collectors.toSet());
         return new ProxyResolutionResult(filters, ingresses, clusterRefs, clusterResolutionResults);
     }
 
     private ClusterResolutionResult discoverProblemsAndBuildResolutionResult(VirtualKafkaCluster cluster,
                                                                              Map<LocalRef<KafkaProxyIngress>, KafkaProxyIngress> ingresses,
                                                                              Map<LocalRef<KafkaService>, KafkaService> services,
                                                                              Map<LocalRef<KafkaProtocolFilter>, KafkaProtocolFilter> filters,
-                                                                             Set<KafkaProxy> proxies) {
-        VirtualKafkaClusterSpec spec = cluster.getSpec();
-        LocalRef<VirtualKafkaCluster> clusterRef = toLocalRef(cluster);
-        Set<DanglingReference> danglingReferences = determineDanglingRefs(ingresses, services, filters, proxies, clusterRef, spec);
-        return new ClusterResolutionResult(cluster, danglingReferences, determineResolvedRefsFalse(ingresses, services, filters));
+                                                                             Set<KafkaProxy> proxies,
+                                                                             boolean considerClusterConditions) {
```

Comment:
passing down a bool is kind of crap, I think what I really want to do is refactor so that the VKC path discovers all the problems with it's dependencies and the KafkaProxy path does that for all VKCs then augments in an additional resolution problem. It feels like the VKC path should not worry about the VKC conditions at all, so this boolean shouldn't exit.

Reply:
have refactored this so that it's now flipped around, instead of the VKC resolution doing a KafkaProxy reconciliation and pulling out the only cluster result, the KafkaProxy resolution is now composed from multiple VKC resolutions and it augments in the additional resolved refs false failure reason.